framework/components/ema-buttons/ema-buttons.twig

```diff
@@ -0,0 +1,48 @@
+{#
+  Accepts:
+    - "modifier" (string): can be "default", "primary", "secondary", "call", "form", "form-primary" or "splash-page" (default: "default")
+    - "type" (string): can be "link" or "button" (default: "button")
+    - "href" (string, if type == "link"): href of the link (default: '#')
+    - "icon" (string): can be empty, "caret-up", "caret-right" or "caret-down"  (default: "")
+    - "label" (string): label of the button (default: "")
+    - "link_label_block": (block): block label of the button (when type == "link")
+    - "button_label_block":(block): block label of the button (when type == "button")
+    - "is_block" (boolean): block-wide button (default: false)
+    - "extra_classes" (string): additional classes to add to the component
+    - "extra_attributes" (array of { name: "", value: "" })
+#}
+
+{# Internal properties #}
+
+{% set _modifier = modifier|default('default') %}
+{% set _type = type|default('button') %}
+{% set _icon = icon|default('') %}
+{% set _label = label|default('') %}
+{% set _href = href|default('#') %}
+{% set _is_block = is_block|default(false) %}
+{% set _css_class = 'ecl-button ecl-button--ema-' ~ _modifier %}
+{% set _extra_attributes = '' %}
+{% set _extra_classes = extra_classes|default('') %}
+
+{# Internal logic - Process properties #}
+
+{% if _is_block == true %}
+  {% set _css_class = _css_class ~ ' ecl-button--block' %}
+{% endif %}
+
+{% if _icon is not empty %}
+  {% set _css_class = _css_class ~ ' ecl-button--' ~ _icon %}
+{% endif %}
+
+{% if _extra_classes is not empty %}
+  {% set _css_class = _css_class ~ ' ' ~ _extra_classes %}
+{% endif %}
+
+{% if extra_attributes is defined and extra_attributes is not empty %}
+  {% for attr in extra_attributes %}
+    {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name ~ '="' ~ attr.value ~ '"' %}
+  {% endfor %}
+{% endif %}
+
+{# Print result #}
+<button class="{{ _css_class }}"{{ _extra_attributes }}>{% block button_label_block _label %}</button>
```

Comment:
You don't even need to have this full twig file; just call ECL button here with `{% include '@ec-europa/ecl-buttons' %}`
(have a look at ema-links.twig).
Then in ema-buttons.config.js add extra_classes for each modifier (as it has been done in ema-links.config.js)